addons/mail/static/src/components/discuss/tests/discuss_inbox_tests.js

```diff
@@ -0,0 +1,325 @@
+odoo.define('mail/static/src/components/discuss/tests/discuss_inbox_tests.js', function (require) {
+'use strict';
+
+const {
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    beforeEach: utilsBeforeEach,
+    nextAnimationFrame,
+    start: utilsStart,
+} = require('mail/static/src/utils/test_utils.js');
+
+
+QUnit.module('mail', {}, function () {
+QUnit.module('components', {}, function () {
+QUnit.module('discuss', {}, function () {
+QUnit.module('discuss_inbox_tests.js', {
+    beforeEach() {
+        utilsBeforeEach(this);
+
+        this.start = async params => {
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                autoOpenDiscuss: true,
+                data: this.data,
+                hasDiscuss: true,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        if (this.widget) {
+            this.widget.destroy();
+        }
+        utilsAfterEach(this);
+    },
+});
+
+QUnit.test('reply: discard on pressing escape', async function (assert) {
+    assert.expect(9);
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                return [{
+                    author_id: [7, "Demo"],
+                    body: "<p>Test</p>",
+                    date: "2019-04-20 11:00:00",
+                    id: 100,
+                    message_type: 'comment',
+                    needaction_partner_ids: [3],
+                    model: 'project.task',
+                    record_name: 'Refactoring',
+                    res_id: 20,
+                }];
+            }
```

Comment:
Prefer setting `data` and just rely on mock server.

Reply:
done